aerospike-core/src/net/connection_pool.rs

```diff
@@ -89,16 +89,14 @@ impl Queue {
             )
             .await;
 
-            if conn.is_err() {
+            let Ok(Ok(conn)) = conn else {
```

Comment:
This is a bugfix to leaked connections. In the ? return case, it would not -- the counter.